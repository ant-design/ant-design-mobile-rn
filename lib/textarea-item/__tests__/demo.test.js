import rnDemoTest from '../../../tests/shared/demoTest';

rnDemoTest('textarea-item');
