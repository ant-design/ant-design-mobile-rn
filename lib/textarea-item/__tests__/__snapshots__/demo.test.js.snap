// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/textarea-item/demo/basic.tsx correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  showsHorizontalScrollIndicator={false}
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View>
      <View>
        <Text
          style={
            Object {
              "backgroundColor": "#f5f5f9",
              "color": "#888888",
              "fontSize": 14,
              "paddingBottom": 9,
              "paddingHorizontal": 15,
              "paddingTop": 15,
            }
          }
        >
          基本
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
              },
              Object {
                "borderBottomWidth": 0.5,
              },
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            clearButtonMode="while-editing"
            editable={true}
            keyboardType="default"
            multiline={true}
            numberOfLines={4}
            onBlur={[Function]}
            onChange={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            placeholder="固定行数"
            style={
              Array [
                Object {
                  "backgroundColor": "#ffffff",
                  "fontSize": 17,
                  "lineHeight": 22,
                  "paddingHorizontal": 8,
                  "textAlignVertical": "top",
                },
                Object {
                  "color": "#000000",
                  "paddingRight": 0,
                },
                Object {
                  "height": 96,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
              },
              Object {
                "borderBottomWidth": 0.5,
              },
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            clearButtonMode="while-editing"
            editable={true}
            keyboardType="default"
            maxLength={100}
            multiline={true}
            numberOfLines={4}
            onBlur={[Function]}
            onChange={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            placeholder="多行带计数"
            style={
              Array [
                Object {
                  "backgroundColor": "#ffffff",
                  "fontSize": 17,
                  "lineHeight": 22,
                  "paddingHorizontal": 8,
                  "textAlignVertical": "top",
                },
                Object {
                  "color": "#000000",
                  "paddingRight": 0,
                },
                Object {
                  "height": 96,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
          />
          <View
            style={
              Array [
                Object {
                  "bottom": 8,
                  "position": "absolute",
                  "right": 8,
                },
              ]
            }
          >
            <Text>
              0
               / 
              100
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
              },
              Object {
                "borderBottomWidth": 0.5,
              },
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            clearButtonMode="while-editing"
            editable={true}
            keyboardType="default"
            multiline={true}
            numberOfLines={4}
            onBlur={[Function]}
            onChange={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            placeholder="高度自适应"
            style={
              Array [
                Object {
                  "backgroundColor": "#ffffff",
                  "fontSize": 17,
                  "lineHeight": 22,
                  "paddingHorizontal": 8,
                  "textAlignVertical": "top",
                },
                Object {
                  "color": "#000000",
                  "paddingRight": 0,
                },
                Object {
                  "height": 96,
                },
                Object {
                  "paddingVertical": 5,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
              },
              Object {
                "borderBottomWidth": 0.5,
              },
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            clearButtonMode="while-editing"
            editable={true}
            keyboardType="default"
            multiline={false}
            numberOfLines={1}
            onBlur={[Function]}
            onChange={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#ffffff",
                  "fontSize": 17,
                  "lineHeight": 22,
                  "paddingHorizontal": 8,
                  "textAlignVertical": "top",
                },
                Object {
                  "color": "#000000",
                  "paddingRight": 0,
                },
                Object {
                  "height": 45,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value="默认带value"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
              },
              Object {
                "borderBottomWidth": 0.5,
              },
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            clearButtonMode="while-editing"
            editable={false}
            keyboardType="default"
            multiline={false}
            numberOfLines={1}
            onBlur={[Function]}
            onChange={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#ffffff",
                  "fontSize": 17,
                  "lineHeight": 22,
                  "paddingHorizontal": 8,
                  "textAlignVertical": "top",
                },
                Object {
                  "color": "#000000",
                  "paddingRight": 0,
                },
                Object {
                  "height": 45,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value="不可编辑 editable = {false}"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
              },
              Object {
                "borderBottomWidth": 0.5,
              },
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            clearButtonMode="never"
            editable={true}
            keyboardType="default"
            multiline={false}
            numberOfLines={1}
            onBlur={[Function]}
            onChange={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            placeholder="不显示清除按钮"
            style={
              Array [
                Object {
                  "backgroundColor": "#ffffff",
                  "fontSize": 17,
                  "lineHeight": 22,
                  "paddingHorizontal": 8,
                  "textAlignVertical": "top",
                },
                Object {
                  "color": "#000000",
                  "paddingRight": 0,
                },
                Object {
                  "height": 45,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
              },
              Object {
                "borderBottomWidth": 0.5,
              },
              Object {
                "position": "relative",
              },
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            clearButtonMode="while-editing"
            defaultValue="报错样式 error={true}"
            editable={true}
            keyboardType="default"
            multiline={false}
            numberOfLines={1}
            onBlur={[Function]}
            onChange={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#ffffff",
                  "fontSize": 17,
                  "lineHeight": 22,
                  "paddingHorizontal": 8,
                  "textAlignVertical": "top",
                },
                Object {
                  "color": "#f50",
                  "paddingRight": 30,
                },
                Object {
                  "height": 45,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
          />
          <View
            accessible={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 18,
                  "top": 12,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#cccccc",
                    "fontFamily": "antoutline",
                    "fontSize": 22,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "color": "#f4333c",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderBottomColor": "#dddddd",
                "borderBottomWidth": 0.5,
                "bottom": 0,
                "height": 1,
                "left": 0,
                "position": "absolute",
                "right": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
